test/userController.test.js

```diff
@@ -58,9 +58,28 @@ afterAll(async () => {
   await mongodb.stop();
 });
 
+beforeEach(async () => {
+  // Delete all users with the same email before creating the test user
+  await User.deleteMany({ email: testUserCredentials.email });
+
+  // Create a test user before each test
+  testUser = await User.create({
+    name: 'ava',
+    username: 'ava',
+    email: 'ava@ava.com',
+    password: 'secret',
+    role: 'user',
+  });
```

Comment:
Since we are creating the test user before each test, we can probably delete the code from `main` branch on line 45 that does the same thing at the very beginning before the tests are run (`beforeAll`)

Reply:
I merged this thread with PS145. Can we close this one? I will merge the full test with the main branch from PS145, after fixing and passing